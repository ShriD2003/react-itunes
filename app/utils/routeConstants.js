export default {
  repos: {
    route: '/',
    props: {
      maxwidth: 500,
      padding: 20
    },
    exact: true
  },

  itunes: {
    route: '/itunes',
    props: {
      maxwidth: 500,
      padding: 20
    },
    exact: true
  },

  trackDetails: {
    route: '/details/:trackId',
    props: {
      width: 50,
      height: 38,
      padding: 20
    },
    exact: true
  }
};
