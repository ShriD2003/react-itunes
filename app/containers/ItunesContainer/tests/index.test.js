/**
 *
 * Tests for ITuneContainer
 *
 */

import React from 'react';
import { fireEvent } from '@testing-library/dom';
import { timeout, renderProvider } from '@utils/testUtils';
import { ItunesContainerTest as ItunesContainer, mapDispatchToProps } from '../index';
import { translate } from '@app/utils';
import { itunesContainerTypes } from '../reducer';

describe('ItunesContainer tests', () => {
  let submitSpy;

  beforeEach(() => {
    submitSpy = jest.fn();
  });

  it('should render and match the snapshot', () => {
    const { baseElement } = renderProvider(<ItunesContainer dispatchSearchSongs={submitSpy} />);
    expect(baseElement).toMatchSnapshot();
  });

  it('should call dispatchClearGridData on empty change in the search box ', async () => {
    const searchSongsSpy = jest.fn();
    const clearGridDataSpy = jest.fn();
    const { getByTestId } = renderProvider(
      <ItunesContainer dispatchClearGridData={clearGridDataSpy} dispatchSearchSongs={searchSongsSpy} />
    );
    fireEvent.change(getByTestId('search-bar'), {
      target: { value: 'a' }
    });
    await timeout(500);

    expect(searchSongsSpy).toHaveBeenCalled();
    fireEvent.change(getByTestId('search-bar'), {
      target: { value: '' }
    });
    await timeout(500);
    expect(clearGridDataSpy).toHaveBeenCalled();
  });

  it('should call dispatchSearchSongs on change and after enter', async () => {
    const searchTerm = 'Alag Aasmaan ';
    const { getByTestId } = renderProvider(<ItunesContainer dispatchSearchSongs={submitSpy} />);
    const searchBar = getByTestId('search-bar');
    fireEvent.change(searchBar, {
      target: { value: searchTerm }
    });
    await timeout(500);
    expect(submitSpy).toHaveBeenCalledWith(searchTerm);

    fireEvent.keyDown(searchBar, {
      key: 'Enter',
      code: 13,
      charCode: 13
    });
    expect(submitSpy).toHaveBeenCalledWith(searchTerm);
  });

  it('should call dispatchSearchSongs on clicking the search icon', async () => {
    const searchTerm = 'Jo Tum Mere Ho';
    const { getByTestId } = renderProvider(<ItunesContainer dispatchSearchSongs={submitSpy} searchTerm={searchTerm} />);
    fireEvent.click(getByTestId('search-bar'));
    await timeout(500);
    expect(submitSpy).toHaveBeenCalledWith(searchTerm);
  });

  it('should  dispatchSearchSongs on update on mount if searchTerm is already persisted', async () => {
    const searchTerm = 'Choo Lo';
    renderProvider(<ItunesContainer searchTerm={searchTerm} gridData={null} dispatchSearchSongs={submitSpy} />);
    await timeout(500);
    expect(submitSpy).toHaveBeenCalledWith(searchTerm);
  });

  it('should validate mapDispatchToProps actions', async () => {
    const dispatchSearchSpy = jest.fn();
    const searchTerm = 'Jo Tum Mere Ho';
    const actions = {
      dispatchSearchSongs: { searchTerm, type: itunesContainerTypes.SEARCH_ITUNES },
      dispatchClearGridData: { type: itunesContainerTypes.CLEAR_GRID_DATA }
    };

    const props = mapDispatchToProps(dispatchSearchSpy);
    props.dispatchSearchSongs(searchTerm);
    expect(dispatchSearchSpy).toHaveBeenCalledWith(actions.dispatchSearchSongs);

    await timeout(500);
    props.dispatchClearGridData();
    expect(dispatchSearchSpy).toHaveBeenCalledWith(actions.dispatchClearGridData);
  });

  it('should render default error message when search goes wrong', () => {
    const defaultError = translate('something_went_wrong');
    const { getByTestId } = renderProvider(<ItunesContainer searchError={defaultError} />);
    expect(getByTestId('itunes-error-message')).toBeInTheDocument();
    expect(getByTestId('itunes-error-message').textContent).toBe(defaultError);
  });

  it('should render default error message when search goes wrong', async () => {
    const { getByTestId } = renderProvider(<ItunesContainer searchError="error" />);
    await timeout(500);
    expect(getByTestId('itunes-error-message'));
  });
});
