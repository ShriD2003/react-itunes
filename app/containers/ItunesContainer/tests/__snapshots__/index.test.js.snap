// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ItunesContainer tests should render and match the snapshot 1`] = `
<body>
  <div>
    <div
      class="sc-gJhJTp jFIsSl"
      maxwidth="100"
      padding="20"
    >
      <div
        class="ant-card ant-card-bordered sc-braxZu hzEVXA css-dev-only-do-not-override-1pg9a38"
        maxwidth="500"
      >
        <div
          class="ant-card-body"
        >
          <p
            class="css-lh9t18 egtqi0h0"
            data-testid="t"
          >
            Search your Songs
          </p>
          <span
            class="ant-input-group-wrapper ant-input-group-wrapper-outlined css-dev-only-do-not-override-1pg9a38 ant-input-search"
          >
            <span
              class="ant-input-wrapper ant-input-group css-dev-only-do-not-override-1pg9a38"
            >
              <input
                class="ant-input css-dev-only-do-not-override-1pg9a38 ant-input-outlined"
                data-testid="search-bar"
                type="text"
                value=""
              />
              <span
                class="ant-input-group-addon"
              >
                <button
                  class="ant-btn css-dev-only-do-not-override-1pg9a38 ant-btn-default ant-btn-icon-only ant-input-search-button"
                  type="button"
                >
                  <span
                    class="ant-btn-icon"
                  >
                    <span
                      aria-label="search"
                      class="anticon anticon-search"
                      role="img"
                    >
                      <svg
                        aria-hidden="true"
                        data-icon="search"
                        fill="currentColor"
                        focusable="false"
                        height="1em"
                        viewBox="64 64 896 896"
                        width="1em"
                      >
                        <path
                          d="M909.6 854.5L649.9 594.8C690.2 542.7 712 479 712 412c0-80.2-31.3-155.4-87.9-212.1-56.6-56.7-132-87.9-212.1-87.9s-155.5 31.3-212.1 87.9C143.2 256.5 112 331.8 112 412c0 80.1 31.3 155.5 87.9 212.1C256.5 680.8 331.8 712 412 712c67 0 130.6-21.8 182.7-62l259.7 259.6a8.2 8.2 0 0011.6 0l43.6-43.5a8.2 8.2 0 000-11.6zM570.4 570.4C528 612.7 471.8 636 412 636s-116-23.3-158.4-65.6C211.3 528 188 471.8 188 412s23.3-116.1 65.6-158.4C296 211.3 352.2 188 412 188s116.1 23.2 158.4 65.6S636 352.2 636 412s-23.3 116.1-65.6 158.4z"
                        />
                      </svg>
                    </span>
                  </span>
                </button>
              </span>
            </span>
          </span>
        </div>
      </div>
    </div>
    <div
      class="sc-elDIKY XBIIX"
      maxwidth="500"
    >
      <div
        class="ant-card ant-card-bordered sc-braxZu fpKLzv css-dev-only-do-not-override-1pg9a38"
        color="grey"
      >
        <div
          class="ant-card-body"
        >
          <p
            class="css-lh9t18 egtqi0h0"
            data-testid="default-message"
          >
            Your Songs will be shown here
          </p>
        </div>
      </div>
    </div>
  </div>
</body>
`;
